src/controllers/aer_controller.hpp

```diff
@@ -1937,12 +1937,14 @@ void Controller::run_circuit_helper(const Circuit &circ,
   else {
     run_circuit_with_sampled_noise(circ, noise, config, shots, state, method,
                                    cache_blocking, result, rng);
+    state.add_reporting_metadata(result);
```

Comment:
Can this just be the existing `add_metadata` function call, and we remove earlier fall to `state.add_metadata(result)` and make sure its done at the end of execution rather than at the start?

Reply:
I see that the only state that actually implements `add_metadata` is `matrix_product_state` so indeed I can move the call to `add_metadata` to the end of execution and then we don't need the method `add_recording_metadata`.